darts/dataprocessing/dtw/_plot.py

```diff
@@ -151,18 +151,19 @@ def plot_alignment(
     time_dim1 = series1._time_dim
     time_dim2 = series2._time_dim
 
-    x_coords1 = np.array(xa1[time_dim1], dtype="datetime64[s]")[path[:, 0]]
-    x_coords2 = np.array(xa2[time_dim2], dtype="datetime64[s]")[path[:, 1]]
+    x_coords1 = np.array(xa1[time_dim1], dtype=xa1[time_dim1].dtype)[path[:, 0]]
+    x_coords2 = np.array(xa2[time_dim2], dtype=xa2[time_dim2].dtype)[path[:, 1]]
 
     y_coords1 = series1.univariate_values()[path[:, 0]]
     y_coords2 = series2.univariate_values()[path[:, 1]]
 
-    x_coords = np.empty(n * 3, dtype="datetime64[s]")
+    x_coords = np.zeros(n * 3, dtype=xa1[time_dim1].dtype)
     y_coords = np.empty(n * 3, dtype=np.float64)
 
     x_coords[0::3] = x_coords1
     x_coords[1::3] = x_coords2
-    x_coords[2::3] = np.datetime64("NaT")
+    if x_coords.dtype == "datetime64[s]":
+        x_coords[2::3] = np.datetime64("NaT")
```

Comment:
Can you change lines 160-166 to the following?
The reason why `x_coords[2::3] = np.nan` failed for integer indexed series, is because only float arrays (not integer) allow NaNs 
```
    if series1.has_datetime_index:
        x_dtype = xa1[time_dim1].dtype
        x_nan = np.datetime64("NaT")
    else:
        x_dtype = np.float64
        x_nan = np.nan
    
    x_coords = np.empty(n * 3, dtype=x_dtype)
    y_coords = np.empty(n * 3, dtype=np.float64)

    x_coords[0::3] = x_coords1
    x_coords[1::3] = x_coords2
    x_coords[2::3] = x_nan
```